bindings/c/test/apitester/TesterMultiTenantCorrectnessWorkload.cpp

```diff
@@ -0,0 +1,258 @@
+			    auto expected = store.getRange(begin, end, results->size() + 10, false);
+			    if (results->size() != expected.size()) {
+				    error(fmt::format("randomGetRangeOp mismatch. expected {} keys, actual {} keys",
+				                      expected.size(),
+				                      results->size()));
+			    } else {
+				    auto expected_kv = expected.begin();
+				    for (auto actual_kv : *results) {
+					    if (actual_kv.key != expected_kv->key || actual_kv.value != expected_kv->value) {
+						    error(fmt::format(
+						        "randomGetRangeOp mismatch. expected key: {} actual key: {} expected value: "
+						        "{:.80} actual value: {:.80}",
+						        fdb::toCharsRef(expected_kv->key),
+						        fdb::toCharsRef(actual_kv.key),
+						        fdb::toCharsRef(expected_kv->value),
+						        fdb::toCharsRef(actual_kv.value)));
+					    }
+					    expected_kv++;
+				    }
+			    }
+			    schedule(cont);
+		    },
+		    tenantId);
+	}
+
+	void randomClearRangeOp(TTaskFct cont, int tenantId) {
+		auto begin = randomKeyName();
+		auto end = randomKeyName();
+		if (begin > end) {
+			std::swap(begin, end);
+		}
+
+		execTransaction(
+		    [begin, end](auto ctx) {
+			    ctx->tx().clearRange(begin, end);
+			    ctx->commit();
+		    },
+		    [this, begin, end, cont]() {
+			    store.clear(begin, end);
+			    schedule(cont);
+		    },
+		    tenantId);
+	}
+
+	void randomClearOp(TTaskFct cont, int tenantId) {
+		int numKeys = Random::get().randomInt(1, maxKeysPerTransaction);
+		auto keys = std::make_shared<std::vector<fdb::Key>>();
+		for (int i = 0; i < numKeys; i++) {
+			keys->push_back(randomExistingKey());
+		}
+		execTransaction(
+		    [keys](auto ctx) {
+			    for (const auto& key : *keys) {
+				    ctx->tx().clear(key);
+			    }
+			    ctx->commit();
+		    },
+		    [this, keys, cont]() {
+			    for (const auto& key : *keys) {
+				    store.clear(key);
+			    }
+			    schedule(cont);
+		    },
+		    tenantId);
+	}
+
+	void randomInsertOp(TTaskFct cont, int tenantId) {
+		int numKeys = Random::get().randomInt(1, maxKeysPerTransaction);
+		auto kvPairs = std::make_shared<std::vector<fdb::KeyValue>>();
+		for (int i = 0; i < numKeys; i++) {
+			kvPairs->push_back(fdb::KeyValue{ randomNotExistingKey(), randomValue() });
+		}
+
+		execTransaction(
+		    [kvPairs](auto ctx) {
+			    for (const fdb::KeyValue& kv : *kvPairs) {
+				    ctx->tx().set(kv.key, kv.value);
+			    }
+			    ctx->commit();
+		    },
+		    [this, kvPairs, cont]() {
+			    for (const fdb::KeyValue& kv : *kvPairs) {
+				    store.set(kv.key, kv.value);
+			    }
+			    schedule(cont);
+		    },
+		    tenantId);
+	}
+
+	void randomOperation(TTaskFct cont) override {
+		OpType txType = (store.size() < initialSize) ? OP_INSERT : (OpType)Random::get().randomInt(0, OP_LAST);
+		int tenantId = Random::get().randomInt(0, numTenants - 1);
+		keyPrefix = fdb::toBytesRef(fmt::format("{}/{}/", workloadId, tenantId));
```

Comment:
I think we should not include tenantId explicitly into the keyPrefix. We should rather rely on the implementation of multitenancy to separate key spaces of different tenants. Tests should not be unnecessarily defensive, because then they will cover fewer possible cases.